### ISSUE TITLE
Migrate to NIP-44 encryption for NWC

### ISSUE THREAD

**Issue by ekzyis** [stackernews/stacker.news#1699]
### Describe the problem you're trying to solve

NWC wallets will add versioning and switch to NIP-44 encryption at some point, see https://github.com/nostr-protocol/nips/pull/1531.

We are about to switch to NDK (#1590), but it doesn't seem to support NIP-44 for NWC specifically yet. However, the [`NDKPrivateSigner` has NIP-44 encryption support](https://github.com/nostr-dev-kit/ndk/blob/d43fecf16a3dfa6adda2e486be0fd2c1cda73e38/ndk/src/signers/private-key/index.ts#L94-L104).

### Describe the solution you'd like

What's needed is to detect which encryption scheme the wallet supports (nip04 or nip44) as specified in https://github.com/nostr-protocol/nips/pull/1531 and then encrypt requests with the corresponding scheme.
 

### Describe alternatives you've considered

n/a

### Additional context

There is some work currently being done in NDK regarding migration to NIP-44 for other NIPs (https://github.com/nostr-dev-kit/ndk/pull/279) but not for NWC afaict.

**Comment by riccardobl** [comment]
i don't think we should try to monkeypatch NDK from our side, if anything, we should PR it to the NDK repo

**Comment by ekzyis** [comment]
Yes, or we fork it and do it ourselves since the maintainers don't seem to be very responsive. This ticket is just about keeping track that we need to find a solution.

**Comment by ekzyis** [comment]
https://github.com/nostr-protocol/nips/pull/1780 was merged.

The spec to negotiate encryption is quite simple for clients to implement (look at `encryption` tag) but we're still encrypting with nip04 because [NDK has no nip44 support for nip47](https://github.com/nostr-dev-kit/ndk/blob/cd78ef6a478db39d20bfe8c03329473359cdbbac/ndk-wallet/src/wallets/nwc/req.ts#L88).

Another reason to switch away from NDK (#2073).

[`snstr` also doesn't support nip44 for nip47](https://github.com/AustinKelsay/snstr/blob/27e882ba8f3b783256b3bce41581d1176ae8378f/src/nip47/client.ts#L709-L716) but it's easier to add there than to NDK.
